forte/data/vocabulary.py

```diff
@@ -0,0 +1,152 @@
+    in the table.
+
+    .. list-table:: Vocabulary Behavior under different settings.
+
+        * - vocab_method
+          - raw (handle outside)
+          - indexing
+          - indexing
+          - one-hot
+          - one-hot
+        * - need_pad
+          - assume False
+          - True
+          - False
+          - True
+          - False
+        * - get_pad_value
+          - None
+          - 0
+          - None
+          - [0,0,0]
+          - None
+        * - inner_mapping
+          - None
+          - 0:pad 1:ele0
+          - 0:ele0
+          - -1:<PAD> 0:ele0
+          - 0:ele0
+        * - element2repr
+          - raise Error
+          - pad->0 ele0->1
+          - ele0->0
+          - <PAD>->[0,0,0] ele0->[1,0,0]
+          - ele0->[1,0,0]
+        * - id2element
+          - raise Error
+          - 0->pad 1->ele0
+          - 0->ele0
+          - -1 -> <PAD> 0->ele0 (be careful)
+          - 0->ele0
+    """
+    PAD_ELEMENT = "<PAD>"
+    UNK_ELEMENT = "<UNK>"
+
+    def __init__(self, method: str, need_pad: bool, use_unk: bool):
+        self.method = method
+        self.need_pad = need_pad
+        self.use_unk = use_unk
+
+        self.element2id_dict: Dict = dict()
+        self.id2element_dict: Dict = dict()
+
+        if method == "one-hot" and need_pad:
+            self.next_id = -1
+        else:
+            self.next_id = 0
+
+        if need_pad:
+            self.add_element(Vocabulary.PAD_ELEMENT)
+
+        if use_unk:
+            self.add_element(Vocabulary.UNK_ELEMENT)
+
+    def add_element(self, element: Hashable):
+        r"""This function will add element to the vocabulary."""
+        if element not in self.element2id_dict:
+            self.element2id_dict[element] = self.next_id
+            self.id2element_dict[self.next_id] = element
+            self.next_id += 1
+
+    def id2element(self, idx: int) -> Hashable:
+        r"""This function will map id to element."""
+        return self.id2element_dict[idx]
+
+    def element2repr(self, element: Hashable) \
+                    -> Union[int, List[int]]:
+        r"""This function will map element to representation."""
+        if self.use_unk:
+            idx = self.element2id_dict.get(element,
+                    self.element2id_dict[Vocabulary.UNK_ELEMENT])
+        else:
+            idx = self.element2id_dict[element]
+
+        if self.method == "indexing":
+            return idx
+        else:
+            vec_size = len(self.element2id_dict)
+            if self.need_pad:
+                vec_size -= 1
+            vec = [0 for _ in range(vec_size)]
+            if idx != -1:
+                vec[idx] = 1
+            return vec
+
+    def __len__(self) -> int:
+        r"""This function return the size of vocabulary."""
+        return len(self.element2id_dict)
+
+    def has_element(self, element: Hashable) -> bool:
+        r"""This function checks whether an element is added to vocabulary."""
```

Comment:
Can you use our google style documentation? Something like this:

https://github.com/asyml/forte/blob/master/forte/data/data_pack.py#L193

Reply:
Got it.